### ISSUE TITLE
`Uncaught TypeError: Object #<Object> has no method 'fire'`

### ISSUE THREAD

**Issue by sebastianzillessen** [ismyrnow/Leaflet.functionaltilelayer#4]
When I try to use your layer with promises I get an error saying: 

```
Uncaught TypeError: Object #<Object> has no method 'fire' leaflet.functional.tilelayer.js:55
(anonymous function) leaflet.functional.tilelayer.js:55
(anonymous function) jquery.js:3069
fire jquery.js:2913 
self.fireWith jquery.js:3025
deferred.(anonymous function) jquery.js:3114
```

Where should `fire` come from and what should it exactly do?

Thanks!

**Comment by ismyrnow** [comment]
Thanks for taking the time to report your issue.

The plugin uses the leaflet defined 'fire' function for firing an event
when the tile starts loading, and again after it's loaded. This is code is
borrowed from leaflet's standard tile layer.

Can you provide a jsfiddle demonstrating the error? If so, I would be happy
to look at it and try to resolve the problem.

**Comment by ismyrnow** [comment]
I was able to fix the problem (see a4c672c). The issue was that inside a promise callback, `this` was no longer referencing the plugin.

Thanks for the heads up!
